deployments/keycloak-security/docker-compose.yml

```diff
@@ -0,0 +1,242 @@
+# Configurable Registries
+#    This compose file allows the composition to be used in multiple environments. Instead of
+#    using hard-coded strings to identify the names of images, variable interpolation is used
+#    to make the registry part of the name configurable at runtime.
+#
+#    If the REGISTRY environment variable is not set, the registry defaults to
+#    Docker Hub (docker.io).
+#
+#    If the REGISTRY variable is set, interpolation on the command line can be used to inject
+#    a registry's address when the Docker service is deployed. the format of the command is:
+#
+#        docker stack deploy -c <(REGISTRY=<registry-url> docker-compose config) cdr
+#
+#    where <registry-url> is changed to be the address of the target registry.
+# --------------------------------------------------------------------------------------------------
+# To deploy to a cloud:
+#    1. Build the images locally.
+#    2. Tag those images to include the target registry address.
+#    3. Push those images.
+#    4. From the cloud environment, execute:
+#
+#        docker stack deploy -c <(REGISTRY=<registry-url> docker-compose config) cdr
+#
+#
+version: '3.7'
+
+networks:
+  default:
+    driver: overlay
+  proxy:
+    driver: overlay
+  security:
+    driver: overlay
+
+volumes:
+  mysql_data:
+    driver: local
+  solr_data:
+    driver: local
+  data:
+
+secrets:
+  aws_s3_secret:
+    file: ./secrets/minio_secret.notsec
+  aws_s3_access:
+    file: ./secrets/minio_access.notsec
+
+configs:
+  ingest_config:
+    file: ./configs/ingest_config.yml
+  s3_config:
+    file: ./configs/s3_config.yml
+  search_config:
+    file: ./configs/search_config.yml
+  store_config:
+    file: ./configs/store_config.yml
+  transform_config:
+    file: ./configs/transform_config.yml
+  keycloak_config:
+    file: ./configs/keycloakJsonConfig.json
+  gateway_config:
+    file: ./configs/gateway_config.yml
+
+services:
+  minio:
+    image: minio/minio
+    hostname: minio
+#    ports:
+#      - target: 9000
+#        published: 8000
+#    #expose
+#    #  - 8080
```

Comment:
❓ Why are these ports commented out? Something to do with the `entrypoint` below?

Reply:
Thanks, I should clean them up a bit. I removed all the published ports since all external traffic should go through the gateway, but I left these commented here as a reference for which ports each service was exposing internally so that it was easier to wire things together.

Docker used to have an "expose" label specifically for this, but when I tried it docker kept printing out "expose isn't needed" messages which made the deploy script confusing to read :joy: